helm/ovn-kubernetes/charts/ovnkube-db/templates/deployment.yaml

```diff
@@ -171,9 +171,29 @@ spec:
 
       # end of container
 
-      nodeSelector:
-        node-role.kubernetes.io/control-plane: ""
-        kubernetes.io/os: "linux"
+      # required to be scheduled on a linux node and only one instance of ovnkube-db pod per node
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+{{- if .Values.global.haveControlPlaneNodes }}
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: Exists
+{{- end }}
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
```

Comment:
I think we can put the whole nodeSelectorTerms in values.yaml, so that users can customize the spec as they wish.
i.e. in values.yaml we will have
```
- matchExpressions:
    - key: node-role.kubernetes.io/control-plane
      operator: Exists
    - key: kubernetes.io/os
       operator: In
       values:
        - "linux"
```
in deployment.yaml:
```
{{- with .Values.nodeSelectorTerms }}
nodeSelectorTerms:
`{{- toYaml . | nindent 8 }}`
{{- end }}
```
example: https://github.com/influxdata/helm-charts/blob/master/charts/telegraf/templates/deployment.yaml#L107

Reply:
fixed